src/Monolog/Handler/ElasticSearchHandler.php

```diff
@@ -112,17 +120,38 @@ public function handleBatch(array $records)
     }
 
     /**
-     * Use Elasticsearch bulk API to send list of documents
-     * @param  array             $documents
+     * Use ElasticSearch bulk API to send list of documents
+     *
+     * @param  array $records
      * @throws \RuntimeException
      */
-    protected function bulkSend(array $documents)
+    protected function bulkSend(array $records)
```

Comment:
I'm not sure if it matters, or if this is against Monolog philosophy or something, but I wanted to mention a potential concern.  It may be prudent to put some kind of limit on the bulk size.

It's possible to make an Elasticsearch cluster very unhappy if you send extremely large bulk requests.  So perhaps a soft limit for bulk size, or automatically send multiple bulk requests if the size is greater than a limit?  Unfortunately the issue is physical size, not number of documents, so it may not be practical to limit.  We recommend bulks in the range of ~50mb to give some context.

Also, it may be against the philosophy of Monolog (e.g. the library just logs whatever you send, and the user is responsible for making sure it's not abusive).  Which would make sense too.

Reply:
I am not sure what @Seldaek thinks, but imho currently it is not necessary, because
> the library just logs whatever you send, and the user is responsible for making sure it's not abusive